lib/src/test/java/org/esbtools/lightbluenotificationhook/NotificationHookTest.java

```diff
@@ -0,0 +1,220 @@
+        JsonDoc.modify(post,new Path("login"),JsonNodeFactory.instance.textNode("blah"),true);
+        
+        List<HookDoc> docs=new ArrayList<>();
+        HookDoc doc=new HookDoc(md,new JsonDoc(pre),new JsonDoc(post),CRUDOperation.UPDATE,"me");
+        docs.add(doc);
+        mediator.req=null;
+        hook.processHook(md,cfg,docs);
+        // No request
+        Assert.assertNull(mediator.req);
+
+        // Update something under personalInfo
+        JsonDoc.modify(post,new Path("personalInfo.company"),JsonNodeFactory.instance.textNode("blah"),true);
+        docs=new ArrayList<>();
+        doc=new HookDoc(md,new JsonDoc(pre),new JsonDoc(post),CRUDOperation.UPDATE,"me");
+        docs.add(doc);
+        mediator.req=null;
+        hook.processHook(md,cfg,docs);
+        // There must be an insertion request
+        Assert.assertNotNull(mediator.req);
+        // Doc is in the insertion req
+        JsonNode notification=mediator.req.getEntityData();
+        Assert.assertNotNull(notification);
+        System.out.println(notification);
+        Assert.assertEquals("user",notification.get("entityName").asText());
+        Assert.assertEquals("5.0.0",notification.get("entityVersion").asText());
+        // The id fields must be there, and nothing else
+        ArrayNode ed=(ArrayNode)notification.get("entityData");
+        Assert.assertEquals(2,ed.size());
+        checkEntityData(ed,"_id","123");
+        checkEntityData(ed,"iduid","345");
+    }
+
+
+    @Test
+    public void testMoreFields() throws Exception {
+        EntityMetadata md=getMd("usermd.json");
+        JsonNode pre=loadJsonNode("userdata.json");
+        // Watch anything under "personalInfo"
+        // Only store id in notification
+        HookConfiguration cfg=new NotificationHookConfiguration(projection("{'field':'personalInfo','recursive':1}"),
+                                                                projection("{'field':'login'}"),
+                                                                false);
+
+        JsonNode post=loadJsonNode("userdata.json");
+        JsonDoc.modify(post,new Path("personalInfo.company"),JsonNodeFactory.instance.textNode("blah"),true);
+        
+        List<HookDoc> docs=new ArrayList<>();
+        HookDoc doc=new HookDoc(md,new JsonDoc(pre),new JsonDoc(post),CRUDOperation.UPDATE,"me");
+        docs.add(doc);
+        mediator.req=null;
+        hook.processHook(md,cfg,docs);
+
+        // There must be an insertion request
+        Assert.assertNotNull(mediator.req);
+        // Doc is in the insertion req
+        JsonNode notification=mediator.req.getEntityData();
+        Assert.assertNotNull(notification);
+        System.out.println(notification);
+        ArrayNode ed=(ArrayNode)notification.get("entityData");
+        Assert.assertEquals(3,ed.size());
+        checkEntityData(ed,"_id","123");
+        checkEntityData(ed,"iduid","345");
+        checkEntityData(ed,"login","bserdar");
+    }
```

Comment:
Can we see a test for object and array node entity data inclusions? For example, will we get a path and value for each item in an array?